README.md

```diff
@@ -1,6 +1,199 @@
+Next, we filter out resources that exist in multiple namespaces. `compose` handles this via merge strategies, but `sync` compares the build version with the namespace versions to select the correct resource. Once sorted, we iterate over the list of resources that differ between the build and the artifact.
+
+> @TODO: (Temporary) Should we include additional technical details here?
+
+### Iterating Through Resources:
+During this phase, the following rules apply:
+- If a resource is deleted from the build, it’s skipped. The developer should handle dependencies manually and bump related resources.
+- If a resource is new, add it to the propagation list.
+- If a resource differs between the build and the artifact, compare versions. If the base version (e.g., base, base-propagation_suffix) differs, add it to the propagation list. Otherwise, copy the resource version from the artifact.
+
+The same process is applied to variables by comparing `group_var` and `vault.yaml` files. Deleted, changed, or added variables will also be propagated.
+
+#### Propagation Map and Timeline:
+To propagate, we first search for the commit where the resource version was set, and a special list of objects called `timeline` (temporary name) is created. Each timeline object contains the commit date, a list of variables to propagate, or a list of resources. The timeline is sorted by date to maintain the correct version sequence as if propagation was applied after each change.
+
+After iterating through the timeline and applying dependent resource versions, the latest state is achieved. At this point, we update the real resource in the build directory.
+
+- If the resource version matches the propagation map, it’s skipped.
+- Otherwise, the version is set as `resource_version-propagated_version`.
+
+---
+
+### Common Propagation Cases (WIP — Convert to Tests?)
+
+#### Case 0:
+As a developer:
+- Clone an existing git repo.
+- Run `plasmactl bump` to update the resource version.
+- Run `plasmactl compose --conflicts-verbosity --skip-not-versioned` to prepare the build.
+- Run `plasmactl bump --sync` to propagate versions of updated resources to dependent resources.
+
+**Expected Outcome:**  
+No bumping or propagation occurs. However, the command should copy propagated versions from the artifact if the base version doesn’t match between the build and artifact.
+
+Test status: **OK**
+
+---
+
+#### Case 1:
+As a developer:
+- Update an existing resource (e.g., change a template, task, dependency).
+- Run `plasmactl bump` to update the resource version.
+- Run `plasmactl compose --conflicts-verbosity --skip-not-versioned`.
+- Run `plasmactl bump --sync` to propagate the new version to dependent resources.
+
+**Expected Outcome:**  
+The new version from the bump (e.g., 1111111111111) should propagate to all dependent resources as `resource_version-1111111111111`.
+
+Test status: **OK**
+
+Additional notes:
+- What if the developer didn't bump the version before propagation?
```

Comment:
`- What if the developer didn't bump the version before propagation? ` > `- If the developer didn't bump the version before propagating, the propagation will be done using current version of modified resources. This may cause dependant roles to not be refreshed.`

+ I suggest we thow a warning when that happen: something like `Warning: the version of some propagated resources was not bumped after updates were made: resource1, resource2, etc`